site/content/blog/carvelizing-helm-chart.md

```diff
@@ -0,0 +1,455 @@
+---
+title: "Carvelizing Helm Chart"
+slug: carvelize-helm-chart
+date: 2022-02-16
+author: Rohit Aggarwal
+excerpt: "Use Carvel to author and consume helm chart "
+image: /img/logo.svg
+tags: ['carvel', 'helm', 'gitops']
+---
+
+In this blog post we will first show you how to wrap and distribute [`Bitnami nginx helm chart`](https://github.com/bitnami/charts/tree/master/bitnami/nginx) as a Carvel package, and then install it on the Kubernetes cluster via PackageInstall CR (via kapp-controller).
+
+## Why should I choose Carvel
+
+Kubernetes configuration takes many forms – plain YAML configurations, Helm charts, ytt templates, jsonnet templates, etc. Software running on Kubernetes lives in many different places, e.g. a Git repository, an archive over HTTP, a Helm repository.
+
+Carvel is a collection of small sharp tools that breaks up the problem into smaller building blocks with clean boundaries. This approach facilitates interoperability with other tools, giving users the option to swap out pieces with tools of their choice. Carvel declarative APIs approach provides a familiar experience for Kubernetes users. This means that package management using Carvel is about declaring desired state via a Package CR and have it be continuously reconciled to the desired state on a Kubernetes cluster.
+
+## Prerequisites
+
+Basic knowledge of imgpkg, kbld, kapp-controller, kapp
+
+[`kbld`](https://carvel.dev/kbld/): kbld incorporates image building and image pushing into your development and deployment workflows.
+
+[`imgpkg`](https://carvel.dev/imgpkg/): A tool to package, distribute and relocate your Kubernetes configuration and dependent OCI images as one OCI artifact: a bundle.
+
+[`kapp`](https://carvel.dev/kapp/): kapp is a cli used to deploy and view groups of Kubernetes resources as “application”.
+
+[`kapp-controller`](https://carvel.dev/kapp-controller/): kapp-controller provides declarative APIs to create, customize, install, and update your Kubernetes applications into packages. It also continuously fetches and reconciles resources to converge into their desired state.
+
+**Installing Carvel Tools**
+
+We'll be using Carvel tools throughout this tutorial, so first we'll install them using [`install.sh`](https://carvel.dev/install.sh) script.
+
+```bash
+$ wget -O- https://carvel.dev/install.sh > install.sh
+
+# Inspect install.sh before running...
+$ sudo bash install.sh
+# Install kapp-controller
+$ kubectl apply -f https://github.com/vmware-tanzu/carvel-kapp-controller/releases/latest/download/release.yml
+```
+
+--------------------------
+
+## Authoring a Carvel Package
+
+To create a package, we need to create two Custom Resources (CRs). We will go through step by step to author an nginx helm chart:
+
+**1. Create PackageMetadata CR**: Package Metadata contains high level information and description about the package. Multiple versions of a package share same package metadata.
```

Comment:
"Package Metadata contains high level information and description about the package"

High level information and description seems  to point towards the same information. Should we go with 
"Package Metadata contains a high level overview of a package , with information like descriptions and a display name."

Reply:
Adoped. Removed from `and description`